Dockerfile

```
@@ -4,7 +4,11 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # Prepare to cross-compile by always running the build stage in the build platform, not the target platform.
-FROM --platform=$BUILDPLATFORM golang:1.21.4 as build-env
+ARG COMPILER_BASE_IMAGE=default_golang
```

`COMPILER_BASE_IMAGE` is for compiling, but `DISTROLESS_BASE_IMAGE` is really for running, perhaps this could be better named?

---

How about...

```suggestion
ARG BUILD_IMAGE=default_golang
```

?